Formula/s/screenpipe.rb

```diff
@@ -0,0 +1,58 @@
+class Screenpipe < Formula
+  desc "Library to build personalized AI powered by what you've seen, said, or heard"
+  homepage "https://github.com/mediar-ai/screenpipe"
+  url "https://github.com/mediar-ai/screenpipe/archive/refs/tags/v0.1.98.tar.gz"
+  sha256 "cb3c8039ecb60d35bacd2b9673db112f907b4a1d3d7c32f49a5e77c0274268ad"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "dbus"
+  depends_on "ffmpeg"
+  depends_on "libxcb"
+  depends_on "openssl@3"
+
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "tesseract"
+  end
+
+  def install
+    features = ["--features", "metal,pipes"] if OS.mac? && Hardware::CPU.arm?
+    system "cargo", "install", *features, *std_cargo_args(path: "screenpipe-server")
+    lib.install "screenpipe-vision/lib/libscreenpipe_#{Hardware::CPU.arch}.dylib" if OS.mac?
+  end
+
+  test do
+    assert_match "Usage", shell_output("#{bin}/screenpipe -h")
+
+    log_file = "#{Dir.home}/.screenpipe/screenpipe.#{Time.now.strftime("%Y-%m-%d")}.log"
+    retries = 3
+    begin
+      pid = fork do
+        exec bin/"screenpipe --debug setup"
+      end
+
+      Timeout.timeout(1800) do
+        loop do
+          sleep 10
+          next unless File.exist?(log_file)
+
+          log_content = File.read(log_file)
+          raise "CLI crashed: 'panic' detected in the log" if log_content.include?("panic")
+          break if log_content.include?("screenpipe setup complete")
+        end
+      end
+    rescue Timeout::Error
+      retries -= 1 if retries.positive?
+      retry if retries.positive?
+      raise "Test timed out after multiple attempts."
```

Comment:
Tests will time out after 15 minutes either way, no need to force that in the test

Reply:
updated the code